Migrate stream package to TypeScript

plouc opened the pull request:
Along with the TypeScript migration, I also made the following changes:
- Replaced `tooltipFormat` with `valueFormat`, and add a `formattedValue` property on each computed datum for potential reuse
- Replaced `legendLabel` and `tooltipLabel` with a single `label` property, I don't think we need 2 different properties for this
- Added support for layers
- Added basic unit tests
- Added support for aria attributes on the SVG container
- Added support for custom tooltip and stack tooltip

wyze reviewed (changes requested):
Looks good. Build is erroring because of formatting in the test file.

plouc commented:
> Looks good. Build is erroring because of formatting in the test file.

@wyze, thx for the review, I fixed the formatting issue and added support for custom tooltips to complete another item from https://github.com/plouc/nivo/issues/192.
